### PR TITLE
feat: Regenerate the newest Compute client

### PR DESCRIPTION
**Pull request by vam-google**
The new client also excludes several legacy GCE sub-services (like `HttpHealthCheck`, in favor oа `HTTPHealthCheck` (yes the names differ only in letters case)). This is consistent with PHP client and also fixes the build on windows (otherwise the classes with the same name but different case would override each other on the case-insensitive windows file system).

### CODE REVIEW

**Comment by vam-google** [comment]
@kolea2 PTAL. We need to push this PR first, before any potential release PRs. This PR has an overwhelming amount of stuff in it, but probably the only files which are actually worth reviewing are the `pom.xml` files. The other changes are caused by the updates in either of the following: GCE AIP itself (most of the chanes), synthtool, gapic-generator, googleapis-discovery (configured to exclude `HttpHealthCheck` stuff).

**Review by kolea2 — APPROVED** [review]
lgtm @chingor13 can you take another quick look since this is a large change?

File: google-cloud-compute/pom.xml [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>com.google.cloud</groupId>
   <artifactId>google-cloud-compute</artifactId>
-  <version>0.0.1-SNAPSHOT</version><!-- {x-version-update:google-cloud-compute:current} -->
+  <version>0.0.1-SNAPSHOT
+  </version><!-- {x-version-update:google-cloud-compute:current} -->
```
Comment (chingor13) [inline google-cloud-compute/pom.xml:9]: This needs to stay a single line for the release automation to correctly update this value

**Review by chingor13 — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: google-cloud-compute/pom.xml (hunk 1 above)
Comment (vam-google) [reply google-cloud-compute/pom.xml]: Thanks for pointing it out. No idea how that happened. Put it back on one line.

**Review by chingor13 — APPROVED** [review]

**Pull request merged**